### PR TITLE
docs: add docs about Backstage Deploy

### PR DESCRIPTION
**Pull request by djamaile**
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by djamaile** [comment]
not stale

**Comment by djamaile** [comment]
@Rugvip 😄

**Review by mitchhentgesspotify — APPROVED** [review]
👍 as the docs are almost identical [to this earlier PR](https://github.com/backstage/backstage/pull/17694)

The only difference is the `npx` command has changed from `npx @backstage/deploy ...` to `npx backstage-deploy`. That's ok, but heads up that it looks a little "suspect" - it's unclear that the owner/collaborator `tsunadev` is a Spotify employee.

We can signal trustworthiness a little bit by publishing it as a user called `spotifydjamailer` perhaps instead? (Also, it could be worth adding a gravatar for the publishing user here - it doesn't have to be Spotify-like, but just having it different from the generated one will be received a little better. Perhaps even [use the picture for the `djamaile` GitHub account](https://github.com/djamaile))

**Comment by djamaile** [comment]
@mitchhentgesspotify yeah the npm account is linked to my Github, which should be public. I can't alter my username but I did reach out to NPM support if they can change it for me. Just updated the profile picture, probably takes while before it is updated?

**Comment by mitchhentgesspotify** [comment]
Sounds good - worst comes to worst, we can always create a new account and publish `0.1.1` as them.

File: docs/deployment/backstage-deploy/aws.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  baseUrl: http://localhost:7007
+```
+
+Now we can deploy our instance!
+
+```bash
+$ npx backstage-deploy aws --stack backstage-poc --create-dockerfile
+```
+
+In the first part of the command, we are specifying that we want to deploy our instance on AWS. With the [`--stack`](https://www.pulumi.com/docs/reference/cli/pulumi_stack/) option, we are providing Pulumi a reference to our container deployment. Furthermore, with the `--create-dockerfile` option, there will be a `Dockerfile` and `.dockerignore` created in the root of the project.
```
Comment (Rugvip) [inline docs/deployment/backstage-deploy/aws.md:169]: Curious about `--create-dockerfile` here, we already ship a `Dockerfile` with `@backstage/create-app`, how come we go this route?

File: microsite/sidebars.json [hunk 2]
```diff
@@ -323,7 +323,12 @@
       "deployment/scaling",
       "deployment/docker",
       "deployment/k8s",
-      "deployment/heroku"
+      "deployment/heroku",
+      {
+        "type": "category",
+        "label": "Deploying on a cloud provider with Backstage Deploy",
```
Comment (Rugvip) [inline microsite/sidebars.json:329]: Perhaps something slightly shorter here + call out status, like "Using Backstage Deploy (alpha)"?

File: docs/deployment/backstage-deploy/aws.md (hunk 1 above)
Comment (djamaile) [reply docs/deployment/backstage-deploy/aws.md:169]: Yeah it's optional for if an user doesn't have a `Dockerfile` provided. I just wanted to display a full command  in action, in reality you could just do `npx backstage-deploy aws` or `npx backstage-deploy --dockerfile <path>` .

File: docs/deployment/backstage-deploy/aws.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+backend:
+-  baseUrl: http://localhost:7007
++  baseUrl: ${BACKSTAGE_HOST}
+  listen:
+    port: 7007
+```
+
+The environment variable `BACKSTAGE_HOST` will be set to the endpoint that AWS Lightsail creates.
+
+In addition, you should create a `app-config.local.yaml`:
```
[the remark this answers is not in the document]
Comment (djamaile) [reply docs/deployment/backstage-deploy/aws.md:147]: Yeah why I recommended this is also for a hacky reason. I use `${BACKSTAGE_HOST}` in the `app-config.yaml` but the env is not set yet when running `yarn build`. Then the building phase fails. But having the `app-config.local.yaml` I guess it merges during the build phase and then it doesn't fail. Is there another way to do this?

File: docs/deployment/backstage-deploy/aws.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+## Configuring the Pulumi CLI
+
+Second, install the [Pulumi CLI](https://www.pulumi.com/docs/get-started/install/).
+
+Then we need to execute the following commands, to set Pulumi up:
+
+```bash
+$ pulumi login --local
+$ export PULUMI_CONFIG_PASSPHRASE="<your-secret>"
```
[the remark this answers is not in the document]
Comment (djamaile) [reply docs/deployment/backstage-deploy/aws.md:122]: Put the notice on top of this

**Comment by ewhite1997** [comment]
@Rugvip @mitchhentgesspotify @djamaile Hi -- is it possible to prioritize this so that we can get this page live ASAP. This is blocking a blog that we want to push live on Monday at the latest to include in the external newsletter on 6/6.

File: docs/deployment/backstage-deploy/aws.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+## Configuring the Pulumi CLI
+
+Second, install the [Pulumi CLI](https://www.pulumi.com/docs/get-started/install/).
+
+Then we need to execute the following commands, to set Pulumi up:
+
+:::tip
+
+Make sure to store your passphrase somewhere save as it is used to encrypt/decrypt your Pulumi config.
```
Comment (Rugvip) [inline docs/deployment/backstage-deploy/aws.md:116]:
```suggestion
Make sure to store your passphrase somewhere safe as it is used to encrypt/decrypt your Pulumi config.
```

**Review by Rugvip — APPROVED** [review]
Alright, let's :shipit:!

Just sneaking in a minor tweak

**Pull request merged**